pages/admin/treatments/index.tsx

```diff
@@ -0,0 +1,286 @@
+        try {
+            await deleteTreatment(modalState.selectTreatment.id);
+            TreatmentNotification.deleteSuccess();
+            await refetch();
+            setModalState({
+                ...modalState,
+                isOpen: false,
+            });
+        } catch (error) {
+            console.error('Error al eliminar el tratamiento', error);
+            TreatmentNotification.deleteError();
+        } finally {
+            setIsSubmitting(false);
+        }
+    }
+
+    function handleEditTreatment(treatment: Treatment) {
+        setModalState({
+            isOpen: true,
+            mode: 'edit',
+            selectTreatment: treatment,
+        });
+    }
+
+    function handleAddTreatment() {
+        if (treatments) {
+            setModalState({
+                isOpen: true,
+                mode: 'create',
+            });
+        }
+    }
+    function handleDeleteTreatment(treatment: Treatment) {
+        setModalState({
+            isOpen: true,
+            mode: 'delete',
+            selectTreatment: treatment,
+        });
+    }
+
+    return (
+        <>
+            <Box
+                p={isSmallScreen ? 3 : 8}
+                m={isSmallScreen ? 3 : 8}
+                bg='white'
+                borderRadius='lg'
+                className='light'
+            >
+                <Box
+                    display='flex'
+                    justifyContent='space-between'
+                    alignItems='center'
+                    mb={6}
+                >
+                    <Text fontSize='2xl' fontWeight='bold'>
+                        Tratamientos
+                    </Text>
+                    <Button
+                        _hover={{ bg: 'blue.500' }}
+                        p={2}
+                        size='lg'
+                        fontWeight='bold'
+                        colorPalette='blue'
+                        onClick={handleAddTreatment}
+                    >
+                        <Plus size={20} />
+                        Crear Tratamiento
+                    </Button>
+                </Box>
+
+                <SearchFilters
+                    filters={filters}
+                    onFilterChange={handleFilterChange}
+                    onSortChange={handleSortChange}
+                    showSort={true}
+                />
+
+                <AdminTable
+                    data={filteredTreatments}
+                    columns={tableColumns}
+                    isLoading={loading}
+                />
+            </Box>
+            <TreatmentModal
+                isOpen={modalState.isOpen}
+                onClose={() => setModalState({ ...modalState, isOpen: false })}
+                mode={modalState.mode}
+                selectedTreatment={modalState.selectTreatment}
+                onTreatmentCreated={() => {
+                    refetch();
+                }}
+                onSubmit={
+                    modalState.mode === 'delete'
```

Comment:
💭 the ternary can come in handy in most situations, but for this specific scenario I find it a bit hard to read, maybe you can separate it in a different function so it's more legible:

```ruby
  const handleOnSubmit = () => {
    if modalState.mode === 'delete' {
      handleConfirmDelete();
      return;
    }

    if modalState.mode === 'edit' {
       handleUpdateTreatment();
       return
    }
 
      handleCreateTreatment();
  }
```

Reply:
if you want to go fancy:

```js
  const handleOnSubmit = () => {
     const actions = {
       edit: handleUpdateTreatment,
       delete: handleConfirmDelete,
       create: handleCreateTreatment,
     }

     actions[modalState.mode]();
  }
```